cmd/kubectl-gadget/main.go

```diff
@@ -81,6 +82,7 @@ func main() {
 		isVersion = isVersion || arg == "version"
 		isDeployUndeploy = isDeployUndeploy || arg == "deploy" || arg == "undeploy"
 		isHelp = isHelp || arg == "--help" || arg == "-h"
+		isCompletion = isCompletion || arg == "completion"
```

Comment:
Some of them we could. Then there is e.g. `isVersion` which is needed separatly

Reply:
I completely reworked this logic to avoid this. Can any of you folks test it again please? (Not only the completion, but other things to be sure I didn't break anything).

Reply:
Would you mind describing a bit more the test you did? Did you have kubeconfig pointing to a cluster? Was Inspektor Gadget deployed?

Reply:
I'm now in full refactoring more. See also https://github.com/inspektor-gadget/inspektor-gadget/issues/2773#issuecomment-2079394906. Let's me see if I can get something working.

Reply:
I updated the PR. I hope this time it works fine. Would you mind checking it again?